src/e2e-tests/src/intTest/java/org/niis/xroad/test/ui/glue/CommonUiStepDefs.java

```diff
@@ -61,6 +61,46 @@ public void doLogin(final String username, final String password) {
                 .click();
     }
 
+    @Given("Error message for incorrect credentials is shown")
+    public void errorMessageIsShown() {
+
+        $(By.xpath("//div[text()[contains(.,'Wrong username or password')]]"))
+                .shouldBe(Condition.visible, Duration.ofSeconds(20));
+    }
+
+     @Given("logout button is being clicked")
+        public void logoutButtonIsClicked() {
+         $(By.xpath("//button[@data-test='username-button']"))
+                        .shouldBe(Condition.visible)
+                        .shouldBe(Condition.enabled)
+                        .click();
+        }
+
+     @Given("{int} seconds of inactivity is passed")
+        public void seconds_of_inactivity_is_passed(Integer int1) {
+        try
+        {
+            Thread.sleep(int1 * 1000);
```

Comment:
Use Selenide.sleep instead.
https://selenide.org/javadoc/current/com/codeborne/selenide/Selenide.html#sleep(long)

Reply:
Or maybe wait for x seconds for login screen to appear?

Reply:
I will try and go with this idea of "waiting until login screen appears", then it won't be sleeping anymore than it needs to. I need to figure out how it would make most sense Then/When -wise.

Reply:
Switched this to Selenide.sleep for now - waiting dynamically is bit cumbersome since i don't have way to integrate expected element to this particular step and have proper When/Then structure.

Reply:
Could you elaborate more what way did not work for you? You can provide non-working test snippet, just to see what did not work

Reply:
Issue is in this `When/Then` -structure, in case i would like to make reusable WHEN `XX seconds of inactivity is passed`, THEN `session timeout popup is appears`. 

Issue is that to separate waiting and what we are waiting, they are in different functions. It would be simple to make something like `when XX seconds of inactivity is passed, session timeout popup appears` but it seems bit bad practice to make such single use function that explicitly wait for some specific elemen  and i don't really know what i should put in that `Then` section if i would do such way.

Reply:
It is completely fine and not a bad practice to make step like this:

When User becomes idle
Then After XX seconds of inactivity,  session timeout popup appears


"User becomes idle' - you can leave step definition empty, or just add Selenide.sleep(1). This is mostly for readability.

"After XX seconds of inactivity,  session timeout popup appears" - all the logic goes here.

Even though reusable step definitions are great, the intent is to have readable scenarios. Thats why making few specialised steps is ok.